### PR TITLE
PipeConsensus: Fix circular replicate in kill -9/restart cases

### PR DESCRIPTION
**Pull request by Pengzna**
as title. Any data synchronized from a replica should not be replicated again.

### CODE REVIEW

File: iotdb-api/pipe-api/src/main/java/org/apache/iotdb/pipe/api/event/dml/insertion/TsFileInsertionEvent.java [hunk 1]
```diff
@@ -34,4 +34,7 @@ public interface TsFileInsertionEvent extends Event, AutoCloseable {
    * @return {@code Iterable<TabletInsertionEvent>} the list of {@link TabletInsertionEvent}
    */
   Iterable<TabletInsertionEvent> toTabletInsertionEvents();
+
+  /** The method is used to prevent circular replication in PipeConsensus */
+  boolean isGeneratedByPipeConsensus();
```
Comment (Caideyipi) [inline iotdb-api/pipe-api/src/main/java/org/apache/iotdb/pipe/api/event/dml/insertion/TsFileInsertionEvent.java:39]: This is pipe api and is exposed to users... Do not include this in interface.

Comment (Pengzna) [reply iotdb-api/pipe-api/src/main/java/org/apache/iotdb/pipe/api/event/dml/insertion/TsFileInsertionEvent.java:39]: ok

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/tsfile/TsFileResource.java [hunk 2]
```diff
@@ -161,6 +161,9 @@ public class TsFileResource {
 
   private ProgressIndex maxProgressIndex;
 
+  /** used to prevent circular replication in PipeConsensus */
+  private boolean isGeneratedByPipeConsensus = false;
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/tsfile/TsFileResource.java:165]: do we need to serialize it?

Comment (Pengzna) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/tsfile/TsFileResource.java:165]: Seems no need. There is no serialization process involved from the generation of tsFileResource by PipeConsensusReceiver to the capture and process of the tsFileInsertionEvent by pipe

Comment (Caideyipi) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/tsfile/TsFileResource.java:165]: > Seems no need. There is no deserialization process involved from the generation of tsFileResource by PipeConsensusReceiver to the capture and process of the tsFileInsertionEvent by pipe; For other processes involving deserialization of TsFileResource, TsFileResource will have a default value of false for `isGeneratedByPipeConsensus`, which does not affect correctness

Now we do the same thing to "isGeneratedByPipe" method and this is also a flag that will not recover. The consequence is that pipe may resend it after restarts, but seems that it does not affect the correctness(The complete correctness may be ensured after load tsfile is enriched with a progressIndex)

**Review by Caideyipi — APPROVED** [review]

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/tsfile/TsFileResource.java (hunk 2 above)
Comment (Pengzna) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/tsfile/TsFileResource.java:165]: @Caideyipi What you said makes sense, I will keep it in mind, thank you!

**Review by SteveYurongSu — APPROVED** [review]

**Review by luoluoyuyu — APPROVED** [review]

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**